java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -2079,6 +2079,34 @@ void teststringReplaceThrowsException() {
     });
   }
 
+  @Test
+  void testStringReplaceWithBackrefs() {
+    try (ColumnVector v = ColumnVector.fromStrings("2020-1-01", "2020-2-02", null);
+         ColumnVector expected = ColumnVector.fromStrings("2020-01-01", "2020-02-02", null);
+         ColumnVector actual = v.stringReplaceWithBackRefs("-([0-9])-", "-0\\1-")) {
```

Comment:
It would be a lot more obvious if we used a simple text in one of the tests instead of just dates. e.g.

``` 
cv: {"*random text*"} 
cv.stringReplaceWithBackRefs("\\*(\w+)\\*", "<b>\\1</b>") -> cv0: "<b>random text</b>"
```

Reply:
Sure, I can add a simpler test as well. I thought it would be good to cover the specific use case that we're using this feature for.